bool pause
---
bool success
string message